client/src/context/AuthContext.js

```diff
@@ -0,0 +1,101 @@
+import React, { useReducer } from 'react';
+import axios from 'axios';
+
+
+
+export const AuthContext = React.createContext();
+
+
+const types = {
+    Set_Is_Loading: "Set_Is_Loading",
+	Set_Logged_In: "Set_Logged_In",
+    Set_User: "Set_User",
+    Set_Isnot_Graduate:'Set_Isnot_Graduate',
+    Set_Error: "Set_Error",
+    Set_Logout:"Set_Logout"
+};
+
+
+const authReducer = (state, action) => {
+	switch (action.type) {
+	case types.Set_Is_Loading:
+		return { ...state, isLoading: action.payload };
+	case types.Set_Error:
+		return { ...state, isLoading: false, error:action.payload };
+	case types.Set_Logged_In:
+		return { ...state, isAuthenticated:true, isLoading: false };
+    case types.Set_User:
+        return { ...state, userName: action.payload, isAuthenticated:true, isLoading: false };
+    case types.Set_Isnot_Graduate:
+        return { ...state, isGraduate:false, isLoading: false }; 
+    case types.Set_Logout:
+        return { ...state, userProfile:null, userName:null, isAuthenticated:false, isLoading: false };
```

Comment:
You aren't setting a specific state with this but rather kind of reseting all of them so I'd name this just simply `Logout`.